psi4/src/psi4/libfock/snLinK.cc

```diff
@@ -0,0 +1,517 @@
+ 
+    timer_off("snLinK: Construct GauXC Factories");
+    
+    // => Part #3: Construct GauXC Integrator <= //
+    timer_on("snLinK: Construct GauXC Integrator");
+
+    radial_points_ = options_.get_int("SNLINK_RADIAL_POINTS");
+    spherical_points_ = options_.get_int("SNLINK_SPHERICAL_POINTS");
+
+    // convert Psi4 fundamental quantities to GauXC 
+    auto gauxc_mol = psi4_to_gauxc_molecule(primary_->molecule());
+    auto gauxc_primary = psi4_to_gauxc_basisset<double>(primary_, basis_tol_, force_cartesian);
+
+    // create snLinK grid for GauXC
+    auto grid_batch_size = options_.get_int("SNLINK_GRID_BATCH_SIZE");
+    auto use_debug_grid = options_.get_bool("SNLINK_USE_DEBUG_GRID");
+
+    auto gauxc_grid = !use_debug_grid ? GauXC::MolGridFactory::create_default_molgrid(
+        gauxc_mol, 
+        pruning_scheme_map[pruning_scheme_],
+        GauXC::BatchSize(grid_batch_size), 
+        radial_scheme_map[radial_scheme_], 
+        GauXC::RadialSize(radial_points_),
+        GauXC::AngularSize(spherical_points_)
+    ) :
+    GauXC::MolGridFactory::create_default_molgrid(
+        gauxc_mol, 
+        pruning_scheme_map[pruning_scheme_],
+        GauXC::BatchSize(grid_batch_size), 
+        radial_scheme_map[radial_scheme_], 
+        GauXC::AtomicGridSizeDefault::UltraFineGrid
+    );
+   
+    // construct load balancer
+    //const size_t quad_pad_value = 1;
+    //auto gauxc_load_balancer = gauxc_load_balancer_factory_->get_instance(*rt, gauxc_mol, gauxc_grid, gauxc_primary, quad_pad_value);
+    auto gauxc_load_balancer = gauxc_load_balancer_factory_->get_instance(*rt, gauxc_mol, gauxc_grid, gauxc_primary);
+    
+    // construct weights module
+    auto gauxc_mol_weights = gauxc_mol_weights_factory_->get_instance();
+
+    // apply partition weights
+    gauxc_mol_weights.modify_weights(gauxc_load_balancer);
+
+    // construct dummy functional
+    // note that the snLinK used by the eventually-called eval_exx function is agnostic to the input functional!
+    const std::string dummy_func_str = "B3LYP";
+    auto spin = options_.get_str("REFERENCE") == "UHF" ? ExchCXX::Spin::Polarized : ExchCXX::Spin::Unpolarized;
+    
+    ExchCXX::Functional dummy_func_key = ExchCXX::functional_map.value(dummy_func_str);
+    ExchCXX::XCFunctional dummy_func = { ExchCXX::Backend::builtin, dummy_func_key, spin };  
+
+    // construct integrator
+    integrator_ = integrator_factory_->get_shared_instance(dummy_func, gauxc_load_balancer); 
+ 
+    timer_off("snLinK: Construct GauXC Integrator");
+    
+    timer_off("snLinK: Setup");
+}
+
+snLinK::~snLinK() {}
+
+void snLinK::print_header() const {
+    if (print_) {
+        outfile->Printf("\n");
+        outfile->Printf("  ==> snLinK: GauXC Semi-Numerical Linear Exchange K <==\n\n");
+        
+        outfile->Printf("    K Execution Space: %s\n", (use_gpu_) ? "Device" : "Host");
```

Comment:
I may be missing something here, but why is this "Device" rather than "GPU"?

Reply:
"Host" and "Device" are another way to colloquially refer to "CPU" and "non-CPU" (the latter can be GPU, or technically something else like FPGAs or TPUs if applicable). Here, "Device" more naturally pairs with "Host" in terms of phrasing, and follows the naming convention that GauXC uses internally. If we were to switch this to "GPU", I'd also switch "Host" to "CPU", though I prefer the "Host"/"Device" naming pair in this context.

Reply:
Got it, I would suggest, however, `"Device (GPU)"` and `"Host (CPU)"`, for the less Tech-savvy and for the sake of clarity

Reply:
You know what, I can accept that. Done and done!